src/MICore/TerminalLauncher.cs

```diff
@@ -106,7 +95,54 @@ protected override string GetProcessArgs()
                 throw new FileNotFoundException(message);
             }
 
-            return string.Format(CultureInfo.InvariantCulture, "{0} \"{1}\" \"{2}\"", launchScript, _title, _initScript);
+            string setEnvironmentScript = GetSetEnvironmentScript();
+            return string.Format(CultureInfo.InvariantCulture, "{0} \"{1}\" \"{2}{3}\"", launchScript, _title, setEnvironmentScript, _initScript);
```

Comment:
You can call me anal, but I don't want to add an extra space there since if there are no environment variables, I don't want the command line to turn into "<SPACE>..." instead of "...". I can add a comment if you think it is confusing.